micro-frontends-with-angular-and-stenciljs.md

```diff
@@ -0,0 +1,150 @@
+
+We are actually doing the [self-contained systems](https://scs-architecture.org/) architecture in my current project. This is a special variation of the micro frontend pattern. Self-contained systems consist of multiple scalable and independent deployable web applications which are preferably linked together via HTML.
+We were facing the need to share layout components such as header and footer. Or building a dashboard, which integrates widgets from several web applications. We don't want to create libraries or something like that, because it is coupling the frameworks and their versions to the applications. The micro frontend pattern is exactly what we need.
+
+The sounds great. Unfortunately, the technologies (i.e. the web frameworks) did not catch up yet. And because Angular, React and VueJs are not ready for this scenario, a lot of manual work is required to compose a micro frontend application.
+
+## Techniques to compose the UI with micro frontends
+
+So let's have a look at our approach for building micro frontends supporting our self-contained systems. We are using Angular in our web applications. I don't want to discuss the framework decision in this article. It was more or less given by the knowledge of the team.
+
+There are multiple ways how we can compose micro frontends together:
+
+* Integrate the HTML via iFrame. This is quite old school, but it works. iFrame integration sounds easy, but I spent a lot of time in the past on sizing and responsiveness working across the applications.
+
+* Load and integrate the HTML via vanilla javascript. This is very painful to get it right. HTML requires javascript files, styles etc. It is a lot of manual work to load everything and make it work.
+
+* A modern approach would be integrating them via WebComponents. Web components can be loaded dynamically at runtime and is a standardized way of doing that with vanilla JS. And there are plenty of frameworks out there to build WebComponents.
+
+We decided to go for WebComponents. WebComponents are standardized and it seems to be the most painless approach. Browser support is not an issue anymore. In the meanwhile, most of the browsers caught up or have been abandoned.
+
+## Finding the right WebComponent framework
+
+After that decision, we needed to find a suitable framework to build the WebComponents. It is possible to build WebComponents with Angular by using Angular Elements. Using Angular would be a good fit because the whole team knows and is developing Angular. Unfortunately, Angular Elements build a complete Angular application into the WebComponents, which results in rather large bundles.
+
+I compared the bundle sizes a while ago. I used [ngx-build-plus](https://github.com/manfredsteyer/ngx-build-plus) to build a single file Angular Elements bundle. Stencil.js outputs a bundle per WebComponent. 
+
+For my example, I got the following values (round in kB):
+
+* Angular: 186 kB
+* Stencil.js: 13kB
```

Comment:
To me, this comparison is not quite clear. What exactly is compared here? A single component? Because further up you say, that Stencil outputs a bundle per component, so I assume we are just looking at the case that you want to share a single component?

Maybe some more explanation above would help to make it clearer also to people not familiar with Stencil / Web Components :-)